### PR TITLE
0.9.0

### PR DESCRIPTION
**Pull request by tylerlong**
## Fix legacy issues

- Update TypeScript and Webpack to latest version
- Replace all appKey/appSecret with clientId/clientSecret
- Replace npm with yarn
- Replace ringcentral with @ringcentral/sdk
- Let e2e test support extension. (previously only username & password)
- Rename "login" to "username" in demo web form
- Upgrade lots of dependencies to latest version
- Make sure all tests pass

## Upgrade to SIP 0.2.x

- Manually merge everything from https://github.com/nerdchacha/ringcentral-web-phone/tree/wp/release/0.9.0-beta
- Manually merge everything changed in the master branch since https://github.com/nerdchacha/ringcentral-web-phone/tree/wp/release/0.9.0-beta
- Make sure all tests pass

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -8,12 +8,10 @@
     "!src/**/*"
   ],
   "dependencies": {
-    "ringcentral": "^3.1.0",
```
Comment (embbnux) [inline bower.json:11]: Do we still need bower?

File: src/session.ts [hunk 2]
```diff
@@ -37,292 +68,279 @@ export interface RCHeaders {
...
+    removeListener?: typeof EventEmitter.prototype.removeListener;
+    /** Remove all event listeners */
+    removeAllListeners?: typeof EventEmitter.prototype.removeAllListeners;
+    /** RingCentral reply with message implementation */
+    replyWithMessage?: typeof replyWithMessage;
+    /**
+     * @internal
+     * Helper method that sends an INFO request to other user agent and then waits for an INFO request from the other user agent
+     */
+    sendInfoAndRecieveResponse?: typeof sendInfoAndRecieveResponse;
```
Comment (embbnux) [inline src/session.ts:170]: a typo of "receive"

File: src/session.ts [hunk 3]
```diff
@@ -894,21 +616,353 @@ function addTrack(this: WebPhoneSession, remoteAudioEle, localAudioEle): void {
...
+        //    - May optionally be set via options passed to Session.invite().
+
+        const sessionDescriptionHandlerOptions =
+            session.sessionDescriptionHandlerOptionsReInvite as SessionDescriptionHandlerOptions;
+        sessionDescriptionHandlerOptions.hold = hold;
+        session.sessionDescriptionHandlerOptionsReInvite = sessionDescriptionHandlerOptions;
+
+        // Send re-INVITE
+        session
+            .invite(options)
```
Comment (embbnux) [inline src/session.ts:907]: Do we still need to set extraHeaders to add contact? https://github.com/ringcentral/ringcentral-web-phone/pull/344/files#diff-96998a2148aa3d52ad3e80303d0ad88fc43a4afd08966b5330a5657021ae52bbR614

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -7,7 +7,7 @@ jobs:
     runs-on: ubuntu-latest # Firefox in it
     strategy:
       matrix:
-        node-version: [14.x]
+        node-version: [18.x]
```
Comment (embbnux) [inline .github/workflows/ci.yml:10]: 16.x for LTS?

File: bower.json (hunk 1 above)
Comment (tylerlong) [reply bower.json:11]: I think we could remove bower support. https://github.com/ringcentral/ringcentral-web-phone/issues/352

File: src/session.ts (hunk 2 above)
Comment (tylerlong) [reply src/session.ts]: Fixed

File: bower.json (hunk 1 above)
Comment (tylerlong) [reply bower.json:11]: Done

File: .github/workflows/ci.yml (hunk 4 above)
Comment (tylerlong) [reply .github/workflows/ci.yml]: Changed

**Review by embbnux — APPROVED** [review]

**Pull request merged**